spec/ParseConfigKey.spec.js

```diff
@@ -0,0 +1,334 @@
+const ParseServer = require('../lib/index').ParseServer;
+const Config = require('../lib/Config');
+const Deprecator = require('../lib/Deprecator/Deprecator');
+
+const doNothing = () => {};
+
+describe('Config Keys', () => {
+  beforeEach(() => {
+    spyOn(Deprecator, 'scanParseServerOptions').and.callFake(doNothing);
+    spyOn(Config, 'put').and.callFake(() => {
+      return {};
+    });
+    // Spy on the console.error method to capture warnings
+    spyOn(console, 'warn');
+  });
+
+  it('should run fine', () => {
+    try {
+      new ParseServer({
+        ...defaultConfiguration,
+      });
+    } catch (err) {
+      fail('Should run without error');
+    }
+  });
+
+  it('should throw 2 warnings for incorrect key names in Config', () => {
+    expect(() => {
+      new ParseServer({
+        ...defaultConfiguration,
+        unknownKeyName: 'unknownValue', // invalid key
+        masterKeyIPs: '', // invalid key
+      });
+    }).toThrowError(
+      'Unknown key(s) found in Parse Server configuration, see other warning messages for details.'
+    );
+    expect(console.error).toHaveBeenCalledWith(
+      'Warning: The following key from ParseServerOptions is not recognized: masterKeyIPs'
+    );
+    expect(console.error).toHaveBeenCalledWith(
+      'Warning: The following key from ParseServerOptions is not recognized: unknownKeyName'
+    );
+  });
+
+  it('should throw incorrect key warning for Schema Config key', () => {
+    expect(() => {
+      new ParseServer({
+        ...defaultConfiguration,
+        schema: {
+          Strict: true, // invalid key
+        },
+      });
+    }).toThrowError(
+      'Unknown key(s) found in Parse Server configuration, see other warning messages for details.'
+    );
+
+    expect(console.error).toHaveBeenCalledWith(
+      'Warning: The following key from SchemaOptions is not recognized: Strict'
+    );
+  });
+
+  it('should throw incorrect key warning for ParseServer Config key', () => {
+    expect(() => {
+      new ParseServer({
+        ...defaultConfiguration,
+        unknownKeyName: 'unknownValue', // invalid key
+      });
+    }).toThrowError(
+      'Unknown key(s) found in Parse Server configuration, see other warning messages for details.'
+    );
+
+    // Check if console.error was called with the expected message
+    expect(console.error).toHaveBeenCalledWith(
+      'Warning: The following key from ParseServerOptions is not recognized: unknownKeyName'
+    );
+  });
+
+  it('should throw incorrect key warning for RateLimitOption Config key', () => {
```

Comment:
All these tests share seem to share much of their code. Would it make senes to just create a single test and run it over an array of options? I believe this whole file can be reduced to a few lines of code.